src/omv/imlib/imlib.h

```diff
@@ -336,75 +336,134 @@ extern const uint16_t ironbow_table[256];
+        uint32_t is_color       :1; \
+        uint32_t is_mutable     :1; \
+        uint32_t /*reserved*/   :4; \
+    };                              \
+    uint32_t pixfmt;                \
+  };                                \
+  uint32_t size; /* for compressed images */ \
+}
+#elif defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#define PIXFORMAT_STRUCT            \
+struct {                            \
+  union {                           \
+    struct {                        \
+        uint32_t /*reserved*/   :4; \
+        uint32_t is_mutable     :1; \
+        uint32_t is_color       :1; \
+        uint32_t is_compressed  :1; \
+        uint32_t is_bayer       :1; \
+        uint32_t /*reserved*/   :8; \
+        uint32_t pixfmt_id      :8; \
+        uint32_t bpp            :8; \
+    };                              \
+    uint32_t pixfmt;                \
+  };                                \
+  uint32_t size; /* for compressed images */ \
 }
-image_bpp_t;
+#else
+#error "Byte order is not defined."
+#endif
 
 typedef struct image {
-    int w;
-    int h;
-    int bpp;
+    uint32_t w;
+    uint32_t h;
+    PIXFORMAT_STRUCT;
     union {
         uint8_t *pixels;
         uint8_t *data;
     };
 } image_t;
 
-void image_init(image_t *ptr, int w, int h, int bpp, void *data);
+void image_init(image_t *ptr, int w, int h, pixformat_t pixfmt, uint32_t size, void *pixels);
 void image_copy(image_t *dst, image_t *src);
 size_t image_size(image_t *ptr);
 bool image_get_mask_pixel(image_t *ptr, int x, int y);
 
-#define IMAGE_IS_MUTABLE(image) \
-({ \
-    __typeof__ (image) _image = (image); \
-    (_image->bpp == IMAGE_BPP_BINARY) || \
-    (_image->bpp == IMAGE_BPP_GRAYSCALE) || \
-    (_image->bpp == IMAGE_BPP_RGB565); \
-})
-
-#define IMAGE_IS_MUTABLE_BAYER(image) \
-({ \
-    __typeof__ (image) _image = (image); \
-    (_image->bpp == IMAGE_BPP_BINARY) || \
-    (_image->bpp == IMAGE_BPP_GRAYSCALE) || \
-    (_image->bpp == IMAGE_BPP_RGB565) || \
-    (_image->bpp == IMAGE_BPP_BAYER); \
-})
-
-#define IMAGE_IS_MUTABLE_BAYER_JPEG(image) \
-({ \
-    __typeof__ (image) _image = (image); \
-    (_image->bpp == IMAGE_BPP_BINARY) || \
-    (_image->bpp == IMAGE_BPP_GRAYSCALE) || \
-    (_image->bpp == IMAGE_BPP_RGB565) || \
-    (_image->bpp == IMAGE_BPP_BAYER) || \
-    (_image->bpp >= IMAGE_BPP_JPEG); \
-})
-
-#define IMAGE_IS_COLOR(image) \
-({ \
-    __typeof__ (image) _image = (image); \
-    (_image->bpp == IMAGE_BPP_RGB565) || \
-    (_image->bpp == IMAGE_BPP_BAYER) || \
-    (_image->bpp >= IMAGE_BPP_JPEG); \
-})
+#define IMAGE_IS_MUTABLE(image)             (image->is_mutable)
+#define IMAGE_IS_COLOR(image)               (image->is_color)
+#define IMAGE_IS_MUTABLE_BAYER(image)       (image->is_mutable || image->is_bayer)
+// TODO: What's this needed for ? This is true for any image.
+#define IMAGE_IS_MUTABLE_BAYER_JPEG(image)  (image->is_mutable || image->is_bayer || image->is_compressed)
```

Comment:
It's basically is valid. So, just replace this with is valid.

Reply:
Removed.